src/ORM/FieldType/DBBoolean.php

```diff
@@ -28,13 +32,20 @@ public function requireField(): void
             'precision' => 1,
             'sign' => 'unsigned',
             'null' => 'not null',
-            'default' => $this->getDefaultValue(),
+            'default' => (int) $this->getDefaultValue(),
             'arrayValue' => $this->arrayValue
         ];
         $values = ['type' => 'boolean', 'parts' => $parts];
         DB::require_field($this->tableName, $this->name, $values);
     }
 
+    public function setValue(mixed $value, null|array|ModelData $record = null, bool $markChanged = true): static
+    {
+        parent::setValue($value);
+        $this->value = $this->convertBooleanLikeValue($value);
```

Comment:
```suggestion
        parent::setValue($this->convertBooleanLikeValue($value), $record, $markChanged);
```
Pass all the args, and don't set the value twice.

Reply:
Updated

Reply:
Still not passing all the args into `parent::setValue()`.

Reply:
Updated